bin/executables/inventory.rb

```diff
@@ -0,0 +1,121 @@
+module Executables
+  class Inventory
+    def initialize(scheduler, job)
+      @scheduler = scheduler
+      @job = job
+      @collector_hash = {}
+    end
+
+    def execute
+      logger.info 'Starting inventory collector'
+      current_time = Time.now.change(sec: 0) # chop off subseconds
+
+      begin
+        inventory_data(current_time)
+      rescue Timeout::Error
+        logger.error "Error collecting inventory for #{current_time}. Collection timed out."
+      rescue StandardError => e
+        logger.fatal "Encountered unhandled exception: #{e.message}."
+        logger.debug e.backtrace
+        @scheduler.shutdown
+        exit(1)
+      end
+      logger.info 'Shutting down inventory collector'
+    end
+
+    private
+
+    def inventory_data(ctime)
+      inv_timestamps = initialize_inventoried_timestamps_with_inventory_for(ctime)
+      # Handle the case where inventory was collected outside of
+      # this process (e.g., meter registration)
+      inv_timestamps.each do |inv|
+        if inv.persisted?
+          @job.unschedule
+        else
+          inv.save
+          collector_queue = active_collectors
+          if collector_queue.empty?
+            logger.debug 'No active datacenters configured. No machine inventory will be collected.'
+          end
+          # Give up after 9 minutes (avoid falling behind by more than one run)
+          Timeout.timeout(9 * 60) do
```

Comment:
It would be better if this hardcoded value is defined as a constant in another file